agent/conf.go

```diff
@@ -161,34 +162,37 @@ func selectNetworkInterface() (string, error) {
 }
 
 type Conf struct {
-	ClusterListenerConfig   ListenerConfig
-	KafkaListenerConfig     ListenerConfig
-	ClusterMembershipConfig cluster.MembershipConf
-	PusherConf              pusher.Conf
-	ControllerConf          control.Conf
-	CompactionWorkersConf   lsm.CompactionWorkerServiceConf
-	FetcherConf             fetcher.Conf
-	FetchCacheConf          fetchcache.Conf
-	GroupCoordinatorConf    group.Conf
-	TxCoordinatorConf       tx.Conf
-	MaxControllerClients    int
+	ClusterListenerConfig     ListenerConfig
+	KafkaListenerConfig       ListenerConfig
+	ClusterMembershipConfig   cluster.MembershipConf
+	PusherConf                pusher.Conf
+	ControllerConf            control.Conf
+	CompactionWorkersConf     lsm.CompactionWorkerServiceConf
+	FetcherConf               fetcher.Conf
+	FetchCacheConf            fetchcache.Conf
+	GroupCoordinatorConf      group.Conf
+	TxCoordinatorConf         tx.Conf
+	MaxControllerClients      int
+	DefaultTopicRetentionTime time.Duration
 }
 
 func NewConf() Conf {
 	return Conf{
-		ClusterMembershipConfig: cluster.NewMembershipConf(),
-		PusherConf:              pusher.NewConf(),
-		ControllerConf:          control.NewConf(),
-		CompactionWorkersConf:   lsm.NewCompactionWorkerServiceConf(),
-		FetcherConf:             fetcher.NewConf(),
-		FetchCacheConf:          fetchcache.NewConf(),
-		GroupCoordinatorConf:    group.NewConf(),
-		TxCoordinatorConf:       tx.NewConf(),
-		MaxControllerClients:    DefaultMaxControllerClients,
+		ClusterMembershipConfig:   cluster.NewMembershipConf(),
+		PusherConf:                pusher.NewConf(),
+		ControllerConf:            control.NewConf(),
+		CompactionWorkersConf:     lsm.NewCompactionWorkerServiceConf(),
+		FetcherConf:               fetcher.NewConf(),
+		FetchCacheConf:            fetchcache.NewConf(),
+		GroupCoordinatorConf:      group.NewConf(),
+		TxCoordinatorConf:         tx.NewConf(),
+		MaxControllerClients:      DefaultMaxControllerClients,
+		DefaultTopicRetentionTime: DefaultTopicRetentionTime,
 	}
 }
 
 const DefaultMaxControllerClients = 10
+const DefaultTopicRetentionTime = 7 * 24 * time.Hour
```

Comment:
Should be name 'DefaultDefaultTopicRetentionTime' as it's the default of the default, not the default.